internal/cmd/server/start_resource_server.go

```diff
@@ -0,0 +1,326 @@
+/*
+Copyright 2023 Red Hat Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in
+compliance with the License. You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under the License is
+distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+implied. See the License for the specific language governing permissions and limitations under the
+License.
+*/
+
+package server
+
+import (
+	"fmt"
+	"log/slog"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/gorilla/mux"
+	"github.com/spf13/cobra"
+
+	"github.com/openshift-kni/oran-o2ims/internal"
+	"github.com/openshift-kni/oran-o2ims/internal/exit"
+	"github.com/openshift-kni/oran-o2ims/internal/logging"
+	"github.com/openshift-kni/oran-o2ims/internal/service"
+	searchmodel "github.com/stolostron/search-v2-api/graph/model"
+)
+
+const (
+	SearchApiUrlPrefix = "search-api-open-cluster-management"
+)
+
+// GraphqlVars contains the GraphQL variables passed to the search API server
+type GraphqlVars struct {
+	Input []struct {
+		Filters []struct {
+			Property string   `json:"property"`
+			Values   []string `json:"values"`
+		} `json:"filters"`
+		Relatedkinds []string `json:"relatedKinds"`
+	} `json:"input"`
+}
```

Comment:
This is very similar to the `SearchInput` struct in the `searchmodel` package imported above. What is the reason for the additional struct?